### PR TITLE
ENH: improved wait status and positioner timeout messages

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

- Refactor TyphosStatusThread to give clearer and more detailed information (may count as API break)
- Change status thread behavior: thread timeout doesn't count as a failure, thread will continue to watch status until it eventually succeeds or fails.
- Adjust timeout messages to be more friendly (a positioner widget timeout doesn't mean your system is broken)
- Adjust timeout messages to disappear once a successful move completes
- More consistently pass through status failure exception messages to the positioner widget
- Give more detailed information about the timeout when it occurs

## Motivation and Context

- Some motors in prod here are timing out and scaring/confusing people consistently
- There are always lots of questions about what the timeout is or where it comes from
- https://jira.slac.stanford.edu/browse/ECS-5454

## How Has This Been Tested?

- Interactively, extensively, using a sim IOC
- Added/adjusted unit tests
- Asked for opinions on the slack channel

## Where Has This Been Documented?

Here only

## Screenshots (if appropriate):
Friendly (doesn't say error) and verbose (says the timeout in tooltip) timeout message
image

Goes away once the move completes
image

Normal errors propagate from the status if available
image

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Code has been checked for threading issues (no blocking tasks in GUI thread)
- [x] Test suite passes locally
- [x] Test suite passes on GitHub Actions
- [x] Ran ``docs/pre-release-notes.sh`` and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Comment by ZLLentz** [comment]
The CI here is going to need a close look...

**Comment by ZLLentz** [comment]
offline tests py39: all passing
online tests py39 conda: one failure, widgets not cleaned up
all other online tests: segfault on something somewhat adjacent to what I worked on here

**Comment by ZLLentz** [comment]
Switching to a simpler ticket in attempt to isolate the problem

**Comment by ZLLentz** [comment]
I'm going to make some more small adjustments here:

- Round the timeout up to the nearest second
- Put the calculation entirely into the mouseover text, not into the main text
- Add a pre-release page

And then I'll solicit some opinions

File: typhos/positioner.py [hunk 1]
```diff
@@ -719,33 +746,81 @@ def alarmKindLevel(self, kind_level: KindLevel):
         if kind_level != self.alarmKindLevel:
             self.ui.alarm_circle.kindLevel = kind_level
 
-    def move_changed(self):
+    def _move_started(self) -> None:
         """Called when a move is begun"""
         logger.debug("Begin showing move in TyphosPositionerWidget")
         self.moving = True
+        self.err_is_timeout = False
```
Comment (tangkong) [inline typhos/positioner.py:753]: Is this used ever?

**Comment by tangkong** [comment]
I only see 5 instances of `TyphosStatusThread` in our org, all of which seem to be covered in this PR.

File: typhos/positioner.py (hunk 1 above)
Comment (ZLLentz) [reply typhos/positioner.py:753]: oops, it's a vestigial boolean!

File: typhos/status.py [hunk 2]
```diff
@@ -42,40 +58,106 @@ class TyphosStatusThread(QThread):
 
     """
     status_started = Signal()
-    status_finished = Signal(object)
-
-    def __init__(self, status, start_delay=0., timeout=10.0, parent=None):
+    status_timeout = Signal()
+    status_finished = Signal(TyphosStatusResult)
+    error_message = Signal(TyphosStatusMessage)
```
[the remark this answers is not in the document]
Comment (ZLLentz) [reply typhos/status.py:82]: This was a good catch- updating the API and not documenting it was not the way

File: typhos/positioner.py [hunk 3]
```diff
@@ -195,18 +196,32 @@ def _clear_status_thread(self):
...
+        thread.error_message.connect(self._set_status_text)
         thread.start()
 
-    def _get_timeout(self, set_position: float, settle_time: float, rescale: float = 1) -> float | None:
+    def _get_timeout(
+        self,
+        set_position: float,
+        settle_time: float,
+        rescale: float = 1,
+    ) -> tuple[int | None, str]:
```
Comment (tangkong) [inline typhos/positioner.py:224]: My perpetual fear of `from __future__ import annotations` makes me fear this, but it seems to have worked.  I should start using it more

**Review by tangkong — APPROVED** [review]
This looks good to me, I like the threading tests.  I wasn't able to actually test it myself outside the suite, but I like the screenshots.  😆

**Pull request merged**